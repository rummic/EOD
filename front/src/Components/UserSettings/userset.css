body{
    overflow-x: hidden;
}

.UserserBox{
    width:100%;

}

.UsersetBox-content{
    margin: auto;
    width:100%;
    position: absolute;
    height: 100%;
    margin-left: 20%;
}

.UsersetBox-form{
    width: 70%;
    margin-top: 5%;
    margin-left: 5%; 
    background-color: white;
    height: 80%;
}

.UsersetBox-form-content{
    padding: 20px;
    margin-left: 5%;
}

.UsersetBox-form-content label{
    text-align: left;
    width: 30%;
    font-size: 30px;
}
.UsersetBox-form-content input{
    text-align: center;
    width: 60%;
    font-size: 30px;
    border:none;
    margin-bottom: 30px;
}

.UsersetBox-form-content input:disabled{
    background-color: white;
}

.UsersetBox-form .UsersetBox-form-button-back{
    width: 100%;
    background-color: #e9ecef;
    border-radius: .25rem;
}

.UsersetBox-form-button{
    padding: 10px;
    margin-right: 20%;
    margin-top: 15px;
    margin-bottom: 15px;
    margin-left: 24%;
    width:50%;
    border: 5px solid rgb(0, 168, 114);
    background: none;
    font-size: 20px;
    font-weight: bold;
    border-radius: 20px;
    color: rgb(0, 168, 114);
}

.UsersetBox-form-button:hover{
    background-color:  rgb(0, 168, 114);
    color:white;
}

#password{
    display: none;
}

.password-button{
    border: 4px solid rgb(0, 168, 114);
    background: none;
    border-radius: 20px;
    color: rgb(0, 168, 114);
    font-size: 18px;
    margin-left: 25%;
}

.password-button:hover{
    color: white;
    background-color: rgb(0, 168, 114)
}

