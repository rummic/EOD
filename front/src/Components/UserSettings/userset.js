
import React, { Component } from 'react';
import { Redirect } from 'react-router-dom';
import Sidebar from '../Navbar/sidebar';
import './userset.css'

const token = sessionStorage.getItem("token");

class userset extends Component {
  constructor(props) {
    super(props);
    this.state = {
      password: "",
      firstName: "",
      lastName: "",
      email: "",
      phoneNumber: "",
      role: "",
      user: []
    };
    this.onChange = this.onChange.bind(this);
  }

  onChange(e) {
    this.setState({ [e.target.name]: e.target.value })
  }

  componentDidMount() {
    fetch('https://localhost:44388/api/Users/' + sessionStorage.getItem("id"), {
      method: 'GET',
      headers: {
        'Content-Type': 'application/json',
        "Authorization": `bearer ${token}`,
      },
    }).then(response => response.json())
      .then(parseJSON => {
        this.setState({
          firstName: parseJSON.value.firstName,
          lastName: parseJSON.value.lastName,
          email: parseJSON.value.email,
          phoneNumber: parseJSON.value.phoneNumber,
          role: parseJSON.value.role,
        })
      })
  }

  update() {
    fetch('https://localhost:44388/api/Users', {
      method: 'PUT',
      headers: {
        'Content-Type': 'application/json',
        "Authorization": `bearer ${token}`
      },
      body: JSON.stringify({
        "login": sessionStorage.getItem('login'),
        "password": this.state.password,
        "firstName": this.state.firstName,
        "lastName": this.state.lastName,
        "email": this.state.email,
        "phoneNumber": this.state.phoneNumber,
        "role": this.state.role
      })
    })
      .then(response => response.json())
      .then(parseJSON => {
        if (parseJSON.hasErrors) {
          document.getElementById("badData").innerHTML = parseJSON.errors;
          document.getElementById("badData").style.color = "red";
        } else {
          alert("Poprawnie zmieniono dane")
          this.props.history.push("/index")
        }
      })
  }

  showInput() {
    document.getElementById("password").style.display = "inline";
    document.getElementById("hideButton").style.display = "none"
  }
  render() {
    if (!sessionStorage.getItem("token")) {
      return (<Redirect to={'/login'} />)
    }
    return (
        <div className="UsersetBox">
        <Sidebar history={this.props.history}/>
           <div className="UsersetBox-content"> 
                <div className="UsersetBox-form">
                <div className="UsersetBox-form-content">
                <label>Imie :</label><input type="text"  disabled value={this.state.firstName} name="firstName" onChange={this.onChange}/>
                <label>Nazwisko :</label><input type="text"  disabled value={this.state.lastName} name="lastName" onChange={this.onChange}/>
                <label>Nick :</label><input type="text"  disabled value={sessionStorage.getItem('login')} name="login" onChange={this.onChange}/>
                <label>Email :</label><input type="text"  value={this.state.email} name="email" onChange={this.onChange}/>
                <label>Hasło :</label>
                <button id="hideButton" variant="primary" onClick={this.showInput} >Zmień hasło</button>
                <input type="password" id="password" placeholder="Podaj hasło" name="password" onChange={this.onChange}/>
                </div>
                <button className="UsersetBox-form-button" onClick={this.update.bind(this)}>Zmień dane</button>
            </div>
        </div>
        </div>
    );
  }
}

export default userset;