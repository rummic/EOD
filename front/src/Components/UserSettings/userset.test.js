import React from "react";
import userset from "./userset";
import { shallow } from 'enzyme';

describe('userset', () => {
    test('Rendering userettings component', () => {
        const wrapper = shallow(<userset />);
        expect(wrapper).toMatchSnapshot();
    });
});  