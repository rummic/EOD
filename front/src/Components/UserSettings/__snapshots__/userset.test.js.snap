// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`userset Rendering userettings component 1`] = `ShallowWrapper {}`;
